packages/logger-plugin/src/action-logger.ts

```diff
@@ -22,14 +23,40 @@ export class ActionLogger {
   }
 
   completed(nextState: any) {
+    if (this.synchronousWorkEnded) {
+      const message = `(async work completed) ${this.getActionLogHeader()}`;
+      this.logWriter.startGroup(message);
+    }
     this.logWriter.logGreen('next state', nextState);
     this.logWriter.endGroup();
+    this.actionCompleted = true;
   }
 
   errored(error: any) {
+    if (this.synchronousWorkEnded) {
```

Comment:
What is `synchronousWorkEnded`?

Reply:
`ActionLogger` instance is created when any action is dispatched. As actions can be sync and async this is a flag that basically says "oh I've completed doing my synchronous job".

Reply:
I think for some reason this is all very complicated, but oh well

Reply:
Need more comments I guess